src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -329,6 +329,8 @@ ClientRequest::process_op(
     }
   }
 
+  co_await ihref.enter_stage<interruptor>(
+    client_pp(*pg).check_already_complete, *this);
```

Comment:
Is it reasonable to merge `check_already_complete` with the follow-up `get_obc`? They are both ordered exclusive, following the same property. Seems to me introducing more mutexes in the critical IO path are unnecessary.

Reply:
I've no problem with merge those two phases, what do you think? @Matan-B

Reply:
We can merge two sequential ordered exclusive stages. The only thing I'm afraid of is readability since we try to squeeze two logical stages into a single one.
Instead of reusing `get_obc` here, what do you think about introducing the merged stage as a new one? We can name it  `check_already_complete_and_get_obc`. Not the best name but future readers will be able to understand why we enter it before checking for already complete.
We should probably also add a comment explaining that the new stage combines two logical stages to avoid mutex usage overhead between sequential ordered exclusive stages.

Reply:
Sorry for the late response, the comment is addressed, I kept the original `get_obc` phase as other requests, like snaptrim, need it. Please take a look, again. Thanks:-)